### PR TITLE
Update Test CSM Plugin to act more like a real plugin

### PR DESCRIPTION
**Pull request by krlberry**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/AlternativeTestCsmModel.cpp [hunk 1]
```diff
@@ -238,6 +238,7 @@ std::string AlternativeTestCsmModel::constructStateFromIsd(const csm::Isd isd){
   for (size_t param_index = 0; param_index < m_param_values.size(); param_index++) {
     state[AlternativeTestCsmModel::PARAM_NAMES[param_index]] = parsedIsd.at(AlternativeTestCsmModel::PARAM_NAMES[param_index]);
   }
+  std::cout << AlternativeTestCsmModel::SENSOR_MODEL_NAME + "\n" + state.dump() << std::endl;
```
Comment (jessemapel) [inline isis/tests/AlternativeTestCsmModel.cpp]: Why is this debug cout here?

File: isis/tests/TestCsmModel.cpp [hunk 2]
```diff
@@ -224,7 +224,7 @@ std::string TestCsmModel::constructStateFromIsd(const csm::Isd isd){
 
   json parsedIsd;
   isdFile >> parsedIsd;
-  // Only extract the first 2 parameters from the file
+  // Only extract the first 3 parameters from the file
```
Comment (jessemapel) [inline isis/tests/TestCsmModel.cpp]: This model only has 2 parameters. I would just change this to say the we only extract the parameters used by the model.

File: isis/tests/TestCsmPlugin.cpp [hunk 3]
```diff
@@ -215,7 +237,17 @@ std::string TestCsmPlugin::convertISDToModelState(
...
+  std::string name = state.value<std::string>("model_name", "");
+
+  if (name == "") {
+    csm::Error::ErrorType aErrorType = csm::Error::INVALID_SENSOR_MODEL_STATE;
+    std::string aMessage = "No model_name key in the model state object.";
+    std::string aFunction = "TestCsmPlugin::getModelNameFromModelState";
+    csm::Error csmErr(aErrorType, aMessage, aFunction);
+    throw(csmErr);
+  }
+  return name;
```
Comment (jessemapel) [inline isis/tests/TestCsmPlugin.cpp:248]: This is not how we are embedding the model name in the state strings. You should instead return the contents of the state string before the first endline. Something like `modelState.substr(0, modelState.find("/n"))` should do it.

**Review by jessemapel — COMMENTED** [review]
Changes look good. I would also modify the test fixture to look for the plugin based on it's static variable name instead of the hardcoded string.

File: isis/tests/TestCsmPlugin.cpp (hunk 3 above)
Comment (krlberry) [reply isis/tests/TestCsmPlugin.cpp:248]: You're right. I was conflating this with how I'd updated the json isds to include the model_name, which I then didn't end up even using, so I'll update this accordingly (I was thinking "ISD", not "state string")

File: isis/tests/TestCsmModel.cpp (hunk 2 above)
Comment (krlberry) [reply isis/tests/TestCsmModel.cpp]: Thanks for catching this. I added one and then removed it.

File: isis/tests/AlternativeTestCsmModel.cpp (hunk 1 above)
Comment (krlberry) [reply isis/tests/AlternativeTestCsmModel.cpp]: Removed, thanks!

File: isis/tests/TestCsmPlugin.cpp [hunk 4]
```diff
@@ -3,10 +3,18 @@
 #include "TestCsmPlugin.h"
 #include "TestCsmModel.h"
 #include "AlternativeTestCsmModel.h"
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
```
Comment (jessemapel) [inline isis/tests/TestCsmPlugin.cpp]: These can be removed now

File: isis/tests/FunctionalTestsCsminit.cpp [hunk 5]
```diff
@@ -318,7 +317,7 @@ TEST_F(DefaultCube, CSMinitSpiceCleanup) {
 TEST_F(DefaultCube, CSMinitSpiceNoCleanup) {
   // Create an ISD that will result in no successful models
   json isd;
-  isd["name"] = "failing_isd";
+  isd["model_name"] = "failing_isd";
```
Comment (jessemapel) [inline isis/tests/FunctionalTestsCsminit.cpp]: The model_name doesn't need to be in the isd at all anymore

Comment (krlberry) [reply isis/tests/FunctionalTestsCsminit.cpp]: Good point! Thanks for pointing this out.

**Review by jessemapel — APPROVED** [review]
Looks good!

**Pull request merged**